src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorPagePropertyActivator.cs

```diff
@@ -108,7 +108,7 @@ internal ViewDataDictionary CreateViewDataDictionary(ViewContext context)
             {
                 valueAccessor = valueAccessors.JsonHelperAccessor;
             }
-            else if (property.PropertyType == typeof(DiagnosticSource))
+            else if (property.PropertyType == typeof(DiagnosticListener))
```

Comment:
This is a breaking change since it's a check for exact equality. This needs to check both `DiagnosticListener` and `DiagnosticSOurce`

Reply:
Ideally with a comment explaining why and unit tests for both situations.